mne/preprocessing/tests/test_xdawn.py

```diff
@@ -52,7 +52,7 @@ def test_xdawn_fit():
     assert_equal(xd.correct_overlap, False)
     # no overlapp correction should give averaged evoked
     evoked = epochs['cond2'].average()
-    assert_array_equal(evoked.data, xd.evokeds_['cond2'].data)
+    assert_array_equal(evoked.data, xd.evokeds_['cond2'])
```

Comment:
I think we should 
* privately store `XDawn._info_ = epochs.info` and `Xdawn._tmin_ = epochs.tmin` when fitting `Xdawn().fit(epochs)` (if not stored already)
* privately store the evoked data in `_Xdawn._evoked['condn']` as a numpy array.
* add a `@property` in `Xdawn` to get an Evoked instance with `mne.EvokedArray(xdawn._evoked['condn'].data, info=self._info_, tmin=self._tmin_)`
* add a `@property` in `XdawnTransformer.evoked` to get the numpy array.

The general idea is to keep the `_XDawn` class as light as possible: i.e. only store variables that are necessary. If it causes an API change, solve it at th `XDawn` level.

Note that `info` is a bit of a dump variables for generic purposes (e.g. we can consult it before doing source reconstruction or filtering to warn classic processing artefact). However, if we start doing "mass multivariate" analyses such (e.g. search light, generalization across time), we'll generate unnecessary overheads and memory loads the pipelines will then be cloned.